comp/otelcol/ddflareextension/impl/extension.go

```diff
@@ -35,36 +37,33 @@ type ddExtension struct {
 	tlsListener net.Listener
 	info        component.BuildInfo
 	debug       extensionDef.DebugSourceResponse
+	configStore configStore
 }
 
 var _ extension.Extension = (*ddExtension)(nil)
 
-// NewExtension creates a new instance of the extension.
-func NewExtension(_ context.Context, cfg *Config, telemetry component.TelemetrySettings, info component.BuildInfo) (extensionDef.Component, error) {
-	ext := &ddExtension{
-		cfg:       cfg,
-		telemetry: telemetry,
-		info:      info,
-		debug: extensionDef.DebugSourceResponse{
-			Sources: map[string]extensionDef.OTelFlareSource{},
-		},
+func (ext *ddExtension) NotifyConfig(_ context.Context, conf *confmap.Conf) error {
+	var cfg *configSettings
+	var err error
+
+	if cfg, err = unmarshal(conf, *ext.cfg.factories); err != nil {
+		fmt.Println(fmt.Errorf("cannot unmarshal the configuration: %w", err))
```

Comment:
why can't we do something like this https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/891f0eadc18fe87b162ee933b2aeb3c4bbf47f6f/extension/opampextension/opamp_agent.go#L293-L298

Reply:
We need to convert to `otelcol.Config` in order to take advantage of `configopaque.String` (redaction of sensitive fields).